### PR TITLE
feat(extractor): split backslash

### PR DESCRIPTION
**Pull request by XiNiHa**
Before this PR, given the string `"class=\"bg-white\""`, the default split extractor extracted `bg-white\` instead of `bg-white`.
This becomes problematic when extracting from SolidJS SSR output, which becomes an issue when using SolidJS with Astro.
This PR fixes the issue by including `\"` in the split token list.

### CODE REVIEW

**Comment by XiNiHa** [comment]
I agree that this can be somewhat opinionated. I fully respect your opinion and if you don't like it, I'll just use it as a custom extractor in my projects :D

File: packages/core/src/extractors/split.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import type { Extractor } from '../types'
 import { isValidSelector } from '../utils'
 
-export const splitCode = (code: string) => code.split(/[\s'"`;=]+/g).filter(isValidSelector)
+export const splitCode = (code: string) => code.split(/[\s'"`;=]|(?:\")+/g).filter(isValidSelector)
```
Comment (antfu) [inline packages/core/src/extractors/split.ts:4]:
```suggestion
export const splitCode = (code: string) => code.split(/[\s'"`;=]|(?:\\["'`])+/g).filter(isValidSelector)
```

**Comment by XiNiHa** [comment]
Okay I guess `\` was not allowed in the rules name anyways XD

**Comment by antfu** [comment]
Ah, we do have some cases that need backslash 🤔

**Comment by XiNiHa** [comment]
```
 ❯ test/preset-mini.test.ts  (7 tests | 1 failed) 272ms
   ❯ test/preset-mini.test.ts > preset-mini > targets
     → expected [ '[content:attr(attr\_content)]' ] to deeply equal []
```
Looks like we should strictly check for `\\["']`?

**Comment by antfu** [comment]
I guess we could do `/\\?[\s'"`;=]+/g

**Comment by antfu** [comment]
Can you add a test for your case `"class=\"bg-white\""` so we won't break the usage accidentally in the future

**Pull request merged**